### PR TITLE
Finish basic api setup

### PR DESCRIPTION
**Pull request by Junyu06**
gender
age
weight
friend
sleep data
heartbeep data

### CODE REVIEW

**Comment by FarWrong** [comment]
2 Things:
Heartbeat API: should have a begin and end with the average telling the average time between: ALSO

Validation for API writes should be heavier, you should check type/validity harder(what if someone writes "2134asdfc" as an age input, need strict policing

**Comment by Junyu06** [comment]
V0.2 is updated. Please review it

Fix v0.1 API's issues:
1. null/none is accessible when a new user is created

2. change the sleep and heart data; it only takes time and average now.

New Function:
1. userinfo API - it return everything about a user to the API by json

Future plan work:
1. add first name and last name file and APIs related to it
2. Create user by using api

**Review by FarWrong — APPROVED** [review]
1. Have approval functionality for friend adding-> You could have it so that each model has the data "pending friend requests" as a stack and adding a friend writes to that stack and approving a friend in pending friend requests is the only way to add a friend.
2.have all user data be retrievable through a single API call-> merge all the api calls (eg there should be a single one that fetches gender, age, etc.

**Pull request merged**
